### PR TITLE
add compare_versions(expression) function

### PR DESCRIPTION
**Pull request by dahorak**
- fix skipif_ocp_version and skipif_ocs_version functions
- add unittest for the compare_versions function

NOTE: it changes the behavior of `skipif_ocp_version` for multiple expressions passed, but since we use everywhere only one expression, it should be safe and it also unify it with `skipif_ocs_version`.

Fixes: https://github.com/red-hat-storage/ocs-ci/issues/5260

Signed-off-by: Daniel Horak <dahorak@redhat.com>

### CODE REVIEW

File: ocs_ci/utility/utils.py [hunk 1]
```diff
@@ -2500,7 +2495,7 @@ def skipif_ocs_version(expressions):
         'True' if test needs to be skipped else 'False'
     """
     expr_list = [expressions] if isinstance(expressions, str) else expressions
-    return any(eval(config.ENV_DATA["ocs_version"] + expr) for expr in expr_list)
+    return any(config.ENV_DATA["ocs_version"] + expr for expr in expr_list)
```
Comment (petr-balogh) [inline ocs_ci/utility/utils.py:2498]:
```suggestion
    return any((config.ENV_DATA["ocs_version"] + expr) for expr in expr_list)
```

File: ocs_ci/utility/utils.py [hunk 2]
```diff
@@ -2476,14 +2476,9 @@ def skipif_ocp_version(expressions):
...
     """
-    skip_this = True
     ocp_version = get_running_ocp_version()
     expr_list = [expressions] if isinstance(expressions, str) else expressions
-    for expr in expr_list:
-        comparision_str = ocp_version + expr
-        skip_this = skip_this and eval(comparision_str)
-    # skip_this will be either True or False after eval
-    return skip_this
+    return any(ocp_version + expr for expr in expr_list)
```
Comment (petr-balogh) [inline ocs_ci/utility/utils.py:2481]:
```suggestion
    return any((ocp_version + expr) for expr in expr_list)
```

File: ocs_ci/utility/utils.py (hunk 1 above)
Comment (dahorak) [reply ocs_ci/utility/utils.py]: good catch :-) ... fixed in the new commit

File: ocs_ci/utility/utils.py (hunk 2 above)
Comment (dahorak) [reply ocs_ci/utility/utils.py]: good catch :-) ... fixed in the new commit

**Review by petr-balogh — DISMISSED** [review]

**Review by ramkiperiy — APPROVED** [review]

**Review by clacroix12 — APPROVED** [review]

File: ocs_ci/utility/tests/test_version.py [hunk 3]
```diff
@@ -63,3 +64,91 @@ def test_compare_from_get_semantic_version():
...
+        "4.1 ===4.1",
+        "Foo version",
+        "<4.1",
+        "==4.1",
+    ],
+)
+def test_compare_versions_wrong_expression(expression):
+    """
+    This test is suppose to test if the compare_versions raises
+    expected exception for wron expression.
```
[the remark this answers is not in the document]
Comment (dahorak) [reply ocs_ci/utility/tests/test_version.py]: Fixed in d25dc68.

**Review by vavuthu — APPROVED** [review]

**Review by Avilir — APPROVED** [review]
LGTM

**Review by ramkiperiy — APPROVED** [review]

**Review by petr-balogh — APPROVED** [review]

**Review by clacroix12 — APPROVED** [review]

**Pull request merged**